src/librawspeed/common/AbstractException.h

```diff
@@ -18,27 +19,33 @@
     Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
 */
 
-#include "io/FileIOException.h"
-#include "common/Common.h" // for _RPT1
-#include <cstdarg>         // for va_end, va_list, va_start
-#include <cstdio>          // for vsnprintf
-#include <string>          // for string
+#pragma once
 
-using namespace std;
+#include "common/Common.h"
+#include <cstdarg>
+#include <cstdio>
+#include <stdexcept>
+#include <string>
 
 namespace RawSpeed {
 
-FileIOException::FileIOException(const string &error)
-    : RawDecoderException(error) {}
-
-void ThrowFIE(const char* fmt, ...) {
+template <typename T>
+[[noreturn]] void __attribute__((format(printf, 1, 2)))
+ThrowException(const char* fmt, ...) {
+  thread_local static char buf[8192];
   va_list val;
   va_start(val, fmt);
-  static char buf[8192];
   vsnprintf(buf, 8192, fmt, val);
   va_end(val);
   writeLog(DEBUG_PRIO_EXTRA, "EXCEPTION: %s\n", buf);
-  throw FileIOException(buf);
+  throw T(buf);
 }
 
+class AbstractException : public std::runtime_error {
+public:
```

Comment:
+1 for introducing a proper base class! But here the term "Abstract" seems really questionable. `std::exception` is also not called `std::abstract_exception`. Maybe just `Exception`? The namespace makes it sufficiently distinct. Or `RawSpeedException`. It would be the name that might show up in client code because someone want's to have it a little more specific than `std::exception`.